src/pkg/golang/build.go

```diff
@@ -301,40 +417,69 @@ func (b *BuildOpts) RegisterFlags(f *flag.FlagSet) {
 }
 
 func (c Environ) build(dirPath string, binaryPath string, pattern []string, opts *BuildOpts) error {
-	args := []string{
-		// Force rebuilding of packages.
-		"-a",
 
+	if err := c.InitCompiler(); err != nil {
+		return err
+	}
+
+	args := []string{
 		"-o", binaryPath,
 	}
+
 	if c.GO111MODULE != "off" && len(c.Mod) > 0 {
 		args = append(args, "-mod", string(c.Mod))
 	}
 	if c.InstallSuffix != "" {
 		args = append(args, "-installsuffix", c.Context.InstallSuffix)
 	}
-	if opts == nil || !opts.EnableInlining {
-		// Disable "function inlining" to get a (likely) smaller binary.
-		args = append(args, "-gcflags=all=-l")
-	}
-	if opts == nil || !opts.NoStrip {
-		// Strip all symbols, and don't embed a Go build ID to be reproducible.
-		args = append(args, "-ldflags", "-s -w -buildid=")
+
+	switch c.Compiler.Identifier {
+	case "go":
```

Comment:
Same thing here, this could be separated from the main logic of the build function to make it more readable.